clmm/theory/ccl.py

```diff
@@ -130,51 +132,64 @@ def _get_einasto_alpha(self, z_cl=None):
 
     def _eval_surface_density(self, r_proj, z_cl):
         a_cl = self.cosmo.get_a_from_z(z_cl)
-        if self.halo_profile_model == 'nfw':
-            return self.hdpm.projected(self.cosmo.be_cosmo, r_proj/a_cl, self.__mdelta_cor,
-                                       a_cl, self.mdef)*self.cor_factor/a_cl**2
-        else:
-            rtmp = np.geomspace(np.min(r_proj)/10., np.max(r_proj)*10., 1000)
-            tmp = self.hdpm.projected(self.cosmo.be_cosmo, rtmp/a_cl, self.__mdelta_cor,
-                                      a_cl, self.mdef)*self.cor_factor/a_cl**2
-            ptf = interp1d(np.log(rtmp), np.log(tmp), bounds_error=False, fill_value=-100)
-            return np.exp(ptf(np.log(r_proj)))
+
+        return self.hdpm.projected(self.cosmo.be_cosmo, r_proj/a_cl, self.__mdelta_cor,
+                                   a_cl, self.mdef)*self.cor_factor/a_cl**2
 
     def _eval_mean_surface_density(self, r_proj, z_cl):
         """"eval mean surface density"""
         a_cl = self.cosmo.get_a_from_z(z_cl)
-        if self.halo_profile_model =='nfw':
-            return self.hdpm.cumul2d(
-                self.cosmo.be_cosmo, r_proj/a_cl, self.__mdelta_cor,
-                self.cosmo.get_a_from_z(z_cl), self.mdef)*self.cor_factor/a_cl**2
-        else:
-            rtmp = np.geomspace(np.min(r_proj)/10., np.max(r_proj)*10., 1000)
-            tmp = self.hdpm.cumul2d(self.cosmo.be_cosmo, rtmp/a_cl, self.__mdelta_cor,
-                                    a_cl, self.mdef)*self.cor_factor/a_cl**2
-            ptf = interp1d(np.log(rtmp), np.log(tmp), bounds_error=False, fill_value=-100)
-            return np.exp(ptf(np.log(r_proj)))
+
+        return self.hdpm.cumul2d(self.cosmo.be_cosmo, r_proj/a_cl, self.__mdelta_cor,
+                                 a_cl, self.mdef)*self.cor_factor/a_cl**2
 
     def _eval_excess_surface_density(self, r_proj, z_cl):
         """"eval excess surface density"""
         a_cl = self.cosmo.get_a_from_z(z_cl)
-        r_cor = r_proj/a_cl
-
-        if self.halo_profile_model =='nfw':
-            return (self.hdpm.cumul2d(self.cosmo.be_cosmo, r_cor, self.__mdelta_cor,
-                                      self.cosmo.get_a_from_z(z_cl), self.mdef)-
-                    self.hdpm.projected(self.cosmo.be_cosmo, r_cor, self.__mdelta_cor,
-                                        self.cosmo.get_a_from_z(z_cl), self.mdef)
-                    )*self.cor_factor/a_cl**2
-        else:
-            return self.eval_mean_surface_density(r_proj, z_cl) \
-                    - self.eval_surface_density(r_proj, z_cl)
 
+        return (self.hdpm.cumul2d(self.cosmo.be_cosmo, r_proj/a_cl, self.__mdelta_cor,
+                                  a_cl, self.mdef) -
+                    self.hdpm.projected(self.cosmo.be_cosmo, r_proj/a_cl, self.__mdelta_cor,
+                                        a_cl, self.mdef))*self.cor_factor/a_cl**2
+
+    def _eval_convergence_core(self, r_proj, z_cl, z_src):
```

Comment:
@hsinfan1996 shouldn't these functions have `self.__mdelta_cor` instead of `self.mdelta`  as input? I guess the correction factor is small enough for the tests to pass, but this should be checked.

Reply:
Fixed.